File: thisboot/src/main/java/com/brageast/blog/thisboot/entity/Reply.java
package com.brageast.blog.thisboot.entity;

import lombok.Builder;
import lombok.Data;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.Date;

/**
 * 回复
 */
@Data
@Builder
@Document
public class Reply {
    private String name;
    private Date startDate;
    private Date upDate;
    private String content;
    /**
     * 多重回复
     */
    private Reply[] replys;
}

File: thisboot/src/main/kotlin/com/brageast/blog/thisboot/entity/Reply.kt
package com.brageast.blog.thisboot.entity

import com.fasterxml.jackson.databind.annotation.JsonSerialize
import com.fasterxml.jackson.databind.ser.std.ToStringSerializer
import org.bson.types.ObjectId
import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document
import java.util.*

/**
 * 回复
 */
@Document(collection = "reply")
data class Reply(
        @Id
        @JsonSerialize(using = ToStringSerializer::class)
        val replyId: ObjectId,
        var name: String? = null,
        var startDate: Date? = null,
        var upDate: Date? = null,
        var content: String? = null,
        // 嵌套回复
        var replys: Array<Reply>? = null
)